### DISCUSSION TITLE
Chrome disallows uploading from Downloads folder on Windows - any remedy?

### DISCUSSION THREAD

**Discussion by PaulVII** [react-dropzone/react-dropzone#1391] [Q&A] [answered]
Chrome restricts access to specific folders depending on the OS (e.g. the user's downloads folder on windows or ~/Library on macOS). I have a hard time understanding when this issue occurs, as many websites have no problem uploading files from those locations. 

In my app using react-dropzone, everything works fine when developing locally, but when deploying it only accepts files from these locations when dropped into the dropzone, but not when selecting them in the dialog. 
The same behaviour can be found on vscode.dev (see https://github.com/microsoft/vscode/issues/144753#issuecomment-1063606119), but it does not occur on other pages (e.g. Google Drive) or even the demo components on https://react-dropzone.js.org/. Is this due to them uploading a file directly via a form? And why should that be considered safer than using the API?

The fact that it works in the react-dropzone docs gives me hope that there is a way to make it work in my app. I'd be glad to hear of your experiences and ideas

**Comment by rolandjitsu** [comment, marked as answer]
@PaulVII thanks for asking here. I think I understand what you're experiencing.

The comment you shared, https://github.com/microsoft/vscode/issues/144753#issuecomment-1063606119, is right. When using the file system access API, the browser may block access to certain locations. But I would expect an error/warning in the console when that happens (we log errors when an error occurs). I'd also have a look at https://developer.mozilla.org/en-US/docs/Web/API/FileSystemHandle/requestPermission, which explains a bit about permissions. Also, keep in mind that for the file system access API to work, you need to be in a secure context.

Our library makes use of the file system access API when opening the file picker, so you should see a similar behaviour (selecting from restricted directories may not work). But note that this API, while it was on by default until https://github.com/react-dropzone/react-dropzone/releases/tag/v14.2.9, it is now disabled by default, so the latest versions, unless you specifically set `useFsAccessApi` to `true`, will use an input to open the file picker, so the behaviour may be different when doing that (it all depends on how the browsers handle that).

Let me know if that makes sense.

**Reply by PaulVII** [reply]
Hi,

thanks a lot for your very thorough and helpful reply! I was on 14.2.3 before, and now both increased the version and explicitly disabled the File System Access API, now the problem no longer occurs. A little sad since using the API feels cleaner, but with this Chrome implementation it feels necessary.

**Reply by rolandjitsu** [reply]
I'm glad I could be of use.
